core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -567,11 +575,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 .field("node", currentNodeId())
                 .field("relocating_node", relocatingNodeId())
                 .field("shard", shardId().id())
-                .field("index", shardId().index().name());
+                .field("index", shardId().index().name())
+                .field("version", version);
+
         if (restoreSource() != null) {
             builder.field("restore_source");
             restoreSource().toXContent(builder, params);
         }
+        if (allocationId != null) {
+            allocationId.toXContent(builder, params);
+        }
```

Comment:
I was weighing the down sides against the upsides and ended up thinking that we end up pulling the entire cluster state with mapping and routing tables for debugging purposes and in that cases I rather have more info than miss something. I'm not sure it adds so much noise. If it helps I can inline as allocation_id and relocating_allocation_id. If you feel strongly about it I'll remove. I just have a gut feeling I'll miss it some day ;)